optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/rostergrid/model/Orientation.java

```diff
@@ -16,48 +16,192 @@
+            element.getElement().style.set("grid-column-end", (endPosition + 1) + "");
+            element.getElement().style.set("--grid-column-end", (startPosition + sizeInGridPixels) + "");
+        }
+
+        @Override
+        void groupScale(final IsElement element, final Long sizeInGridPixels, final Viewport viewport, final Long offsetInScreenPixels) {
+            Long startPosition = getGroupStartPosition(element);
+            element.getElement().style.set("grid-row-end", (startPosition + sizeInGridPixels + 1) + "");
+            element.getElement().style.set("--grid-row-end", (startPosition + sizeInGridPixels) + "");
         }
 
         @Override
         public Long getSize(final IsElement element) {
-            final String size = element.getElement().style.width.asString();
-            return size.isEmpty() ? 0 : Long.parseLong(size.substring(0, size.length() - 2));
+            return getEndPosition(element) - getStartPosition(element);
+        }
+
+        public Long getGroupSize(final IsElement element) {
+            return getGroupEndPosition(element) - getGroupStartPosition(element);
+        }
+
+        private Long getStartPosition(final IsElement element) {
+            final String position = element.getElement().style.get("--grid-column-start");
+            return position == null || position.isEmpty() ? 0 : Long.parseLong(position);
+        }
+
+        private Long getEndPosition(final IsElement element) {
+            final String position = element.getElement().style.get("--grid-column-end");
+            return position == null || position.isEmpty() ? 0 : Long.parseLong(position);
+        }
+
+        private Long getGroupStartPosition(final IsElement element) {
+            final String position = element.getElement().style.get("--grid-row-start");
+            return position == null || position.isEmpty() ? 0 : Long.parseLong(position);
+        }
+
+        private Long getGroupEndPosition(final IsElement element) {
+            final String position = element.getElement().style.get("--grid-row-end");
+            return position == null || position.isEmpty() ? 0 : Long.parseLong(position);
         }
     };
 
+    private static Long clamp(Long value, Long min, Long max) {
+        if (value < min) {
+            return min;
+        } else if (value > max) {
+            return max;
+        }
+        return value;
+    }
+
+    abstract void absGroupPosition(final IsElement element, final Long positionInGridPixels, final Viewport viewport, final Long offsetInScreenPixels);
+
+    abstract void groupPosition(final IsElement element, final Long positionInGridPixels, final Viewport viewport, final Long offsetInScreenPixels);
+
+    // Use this for headers
+    abstract void absPosition(final IsElement element, final Long positionInGridPixels, final Viewport viewport, final Long offsetInScreenPixels);
```

Comment:
Great comment. Maybe make this a javadoc so it shows up in an IDE when someone is trying to figure out which method to call?